src/common.rs

```diff
@@ -71,12 +71,12 @@ pub(crate) const SOCK_CLOEXEC: SockFlag = SockFlag::SOCK_CLOEXEC;
 #[cfg(not(target_os = "linux"))]
 pub(crate) const SOCK_CLOEXEC: SockFlag = SockFlag::empty();
 
-#[cfg(target_os = "linux")]
+#[cfg(all(feature = "abstract_socket", target_os = "linux"))]
```

Comment:
Do we really need feature toggle for this?
Seems like it's easy to support both (containerd supports both options depending on whether there is `\x00` returned from shim start call).
How do you feel about something like (pseudo):
```rust
    let bytes = host.as_bytes();
    if bytes[0] == 0b00 {
        UnixAddr::new_abstract(host[1..])
    } else {
        UnixAddr::new(host)
    }
```

Reply:
I don’t particularly like to use feature too. I tried to extend the Domain enum and control the socket type by the address pattern passed in:

- vsock://xxx:xx -> vsock
- unix:///run/aaa.sock -> unix domain sock
- /run/abc.sock -> abstract sock